File: app/src/main/java/com/samsalek/foodbankandroid/retrofit/RetrofitService.java
package com.samsalek.foodbankandroid.retrofit;

import retrofit2.Retrofit;
import retrofit2.converter.jackson.JacksonConverterFactory;

public class RetrofitService {

    private Retrofit retrofit;

    private final String ip = "192.168.1.20";
    private final String port = "8080";

    public RetrofitService() {
        init();
    }

    private void init() {
        retrofit = new Retrofit.Builder()
                        .baseUrl("http://" + ip + ":" + port)
                        .addConverterFactory(JacksonConverterFactory.create())
                        .build();
    }

    public Retrofit getRetrofit() {
        return retrofit;
    }
}

File: app/src/main/java/io/github/samsalmag/foodbankandroid/retrofit/RetrofitService.kt
package io.github.samsalmag.foodbankandroid.retrofit

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import retrofit2.Retrofit
import retrofit2.converter.jackson.JacksonConverterFactory

class RetrofitService {

    private val objectMapper: ObjectMapper = ObjectMapper()
                                                .registerModule(JavaTimeModule())   // Add module to serialize/deserialize java.time

    private val ip = "192.168.1.20"
    private val port = "8080"

    fun get(): Retrofit {
        return Retrofit.Builder()
            .baseUrl("http://$ip:$port")
            .addConverterFactory(JacksonConverterFactory.create(objectMapper))
            .build()
    }
}
